### PR TITLE
Small cleanups.

### PR DESCRIPTION
**Pull request by anntzer**
Use enumerate(); use decorators.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2967,7 +2967,6 @@ def get_next_color():
         slices = []
         autotexts = []
 
-        i = 0
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: this variable is unused.

File: examples/animation/animate_decay.py [hunk 2]
```diff
@@ -14,9 +14,7 @@
 
 
 def data_gen(t=0):
-    cnt = 0
-    while cnt < 1000:
-        cnt += 1
+    for cnt in range(1000):
         t += 0.1
```
Comment (QuLogic) [inline examples/animation/animate_decay.py]:
```suggestion
        t = cnt / 10
```
(because of floating point addition errors)

Comment (anntzer) [reply examples/animation/animate_decay.py]: fixed, thanks

**Comment by dstansby** [comment]
Needs another rebase?

**Comment by anntzer** [comment]
done

**Pull request merged**